state/executor.go

```diff
@@ -334,17 +337,17 @@ var emptyFrom = types.Address{}
 
 // Write writes another transaction to the executor
 func (t *Transition) Write(txn *types.Transaction) error {
-	var err error
-
-	if txn.From == emptyFrom &&
-		(txn.Type == types.LegacyTx || txn.Type == types.DynamicFeeTx) {
+	if txn.From() == emptyFrom &&
+		(txn.Type() == types.LegacyTx || txn.Type() == types.DynamicFeeTx) {
```

Comment:
And what about access list transactions? Should we change this condition like this maybe:

```suggestion
	if txn.From() == emptyFrom && txn.Type() != types.StateTx{
```